File: Backend/Models/Meal.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MealMate.Api.Models
{
    public class Meal
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; } = string.Empty;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }

        public string? Category { get; set; } = string.Empty;

        public string? ImageUrl { get; set; } = string.Empty;

        public bool IsVeg { get; set; }

        public bool IsAvailable { get; set; } = true;

        [Required]
        public Guid AgentId { get; set; }

        [ForeignKey("AgentId")]
        public User? Agent { get; set; }

        // ── Nutrition & detail fields ──
        public int Calories { get; set; }
        public int Protein { get; set; }       // grams
        public int Carbs { get; set; }         // grams
        public int Fat { get; set; }           // grams
        public int Fiber { get; set; }         // grams
        public string? SpiceLevel { get; set; } = "Medium";   // Mild | Medium | Hot
        public string? PrepTime { get; set; } = "25 min";
        public string? PortionSize { get; set; } = "350g";
        public string? Ingredients { get; set; } = string.Empty;  // comma-separated
        public string? Allergens { get; set; } = string.Empty;    // comma-separated
    }
}

File: Backend-Laravel/app/Models/Meal.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

/**
 * Maps to the "Meals" table shared with the .NET/EF Core backend
 * (see Backend/Models/Meal.cs). Column names intentionally match the
 * existing EF Core schema rather than Laravel snake_case convention.
 */
class Meal extends Model
{
    protected $table = 'Meals';
    protected $primaryKey = 'Id';
    public $incrementing = false;
    protected $keyType = 'string';
    public $timestamps = false;

    protected $fillable = [
        'Id', 'Name', 'Description', 'Price', 'Category', 'ImageUrl',
        'IsVeg', 'IsAvailable', 'AgentId',
        'Calories', 'Protein', 'Carbs', 'Fat', 'Fiber',
        'SpiceLevel', 'PrepTime', 'PortionSize', 'Ingredients', 'Allergens',
        'ReviewsJson', 'Rating', 'ReviewCount',
    ];

    protected $casts = [
        'IsVeg' => 'boolean',
        'IsAvailable' => 'boolean',
        'Price' => 'decimal:2',
        'Rating' => 'float',
    ];

    protected static function booted(): void
    {
        static::creating(function (Meal $meal) {
            if (empty($meal->{$meal->getKeyName()})) {
                $meal->{$meal->getKeyName()} = (string) Str::uuid();
            }
        });
    }

    public function agent()
    {
        return $this->belongsTo(User::class, 'AgentId', 'Id');
    }
}
